src/app/ReadHandler.cpp

```diff
@@ -545,11 +544,7 @@ CHIP_ERROR ReadHandler::ProcessSubscribeRequest(System::PacketBufferHandle && aP
     ReturnLogErrorOnFailure(subscribeRequestParser.GetMinIntervalSeconds(&mMinIntervalFloorSeconds));
     ReturnLogErrorOnFailure(subscribeRequestParser.GetMaxIntervalSeconds(&mMaxIntervalCeilingSeconds));
 
-    // TODO: Use GetSecureRandomData to generate subscription id
-    // it needs #include <support/crypto/CHIPRNG.h>, but somehow CHIPRNG.h is missing
-    // err = Platform::Security::GetSecureRandomData((uint8_t *) &mSubscriptionId, sizeof(mSubscriptionId));
-    // SuccessOrExit(err);
-    mSubscriptionId = GetRandU64();
+    ReturnLogErrorOnFailure(Crypto::DRBG_get_bytes(reinterpret_cast<uint8_t *>(&mSubscriptionId), sizeof(mSubscriptionId)));
```

Comment:
Are there any invalid values of subscription ID?

Reply:
It was promised here that the change to 32-bit subscription ID would be addressed as part of #9520:

https://github.com/project-chip/connectedhomeip/pull/9510#discussion_r706361185

Reply:
It looks that the SubscriptionId type should be addressed as a separate PR.